apps/server/src/controllers/integrationController.ts

```diff
@@ -186,6 +186,15 @@ export function dispatchFromAdapter(type: string, payload: unknown, source?: 'os
       return { topic: 'timer', payload: timer };
     }
 
+    case 'update-duration': {
+      try {
+        const parsedPayload = JSON.parse(payload.toString());
+        PlaybackService.updateEventDuration(parsedPayload.id, parsedPayload.duration);
```

Comment:
what would the payload look like here?
`{id: string, duration: number)`?

We should probably attempt parsing and validating before it reaches the application domain
Can you show me how the OSC message is formed?

Reply:
that's a fair point, this is the OSC message
`{ eventId: '5a6e1', newDuration: 600 }`

I have a python script that allows me to send an OSC message if you want to take a look

```
from pythonosc import udp_client
import json

client = udp_client.SimpleUDPClient(
    "127.0.0.1", 8888)  # Adjust with your IP and Port

# Create the JSON object
data = {
    "id": "5a6e1",
    "duration": 600,
}

# Stringify the JSON object
json_string = json.dumps(data)

# You can put any value you want here
client.send_message("/ontime/update-duration", json_string)
# client.send_message("/ontime/get-timer", "5a6e1")
```

What should be the behaviour if its wrongly formatted/validate? just throw an error?

Reply:
I think throwing an error is sane. If the data is incorrect, somehow it is a no-op for us

The adapters should handle it
https://github.com/cpvalente/ontime/blob/master/apps/server/src/adapters/OscAdapter.ts#L40
https://github.com/cpvalente/ontime/blob/master/apps/server/src/adapters/WebsocketAdapter.ts#L126

making an object can be tricky in most software (I think)
Do you think it would be possible (and better) to do something like `/ontime/update-duration/<eventId> value`?

Similar to what the user described in the issue
https://github.com/cpvalente/ontime/issues/347#issuecomment-1514883828